server/taskflows/hpccloud/taskflow/openfoam/tutorial.py

```diff
@@ -0,0 +1,197 @@
+    task.logger.info('OpenFOAM job created: %s' % job['_id'])
+    task.taskflow.logger.info('OpenFOAM job created.')
+
+    task.taskflow.set_metadata('jobs', [job])
+
+    # Create a copy DockerOpenFOAM_run.sh file over
+    filename = 'DockerOpenFOAM_run.sh'
+
+    target_dir = job_directory(cluster, job)
+    target_path = os.path.join(target_dir, filename)
+
+    # Expose the job directory
+    task.taskflow.set_metadata('dataDir', target_dir)
+
+    submit_open_foam_job.delay(cluster, job,  *args, **kwargs)
+
+
+@cumulus.taskflow.task
+def submit_open_foam_job(task, cluster, job, *args, **kwargs):
+    girder_token = task.taskflow.girder_token
+
+    # Now download and submit job to the cluster
+    task.logger.info('Uploading input files to cluster.')
+    download_job_input_folders(cluster, job, log_write_url=None,
+                        girder_token=girder_token, submit=False)
+    task.logger.info('Uploading complete.')
+
+    task.taskflow.logger.info('Submitting job to cluster.')
+    params = {}
+
+    parallel_environment \
+        = parse('config.parallelEnvironment').find(cluster)
+
+    if parallel_environment:
+        parallel_environment = parallel_environment[0].value
+        params['parallelEnvironment'] = parallel_environment
+
+    params['numberOfSlots'] = 1
+
+    job_output_dir = get_cluster_job_output_dir(cluster)
+    if job_output_dir:
+        params['jobOutputDir'] = job_output_dir
+
+    job['params'] = params
+
+    submit_job(cluster, job, log_write_url=None,
+                          girder_token=girder_token, monitor=False)
+
+    monitor_open_foam_job.delay(cluster, job, *args, **kwargs)
+
+
+@cumulus.taskflow.task
+def monitor_open_foam_job(task, cluster, job, *args, **kwargs):
+    task.logger.info('Monitoring job on cluster.')
+    girder_token = task.taskflow.girder_token
+
+    task.taskflow.on_complete(monitor_job) \
+        .run(upload_output.s(cluster, job, *args, **kwargs))
+
+    task.taskflow.run_task(
+        monitor_job.s(cluster, job, girder_token=girder_token))
+
+@cumulus.taskflow.task
+def upload_output(task, cluster, job, *args, **kwargs):
+    task.taskflow.logger.info('Uploading results from cluster')
+
+    # Refresh state of job
+    client = create_girder_client(
+            task.taskflow.girder_api_url, task.taskflow.girder_token)
+    job = client.get('jobs/%s' % job['_id'])
+
+    output_folder_id = parse('output.folder.id').find(kwargs)
+    if output_folder_id:
+        task.taskflow.logger.info('Exporting to folder %s' % output_folder_id)
+        output_folder_id = output_folder_id[0].value
+        job['output'] = [{
+            'folderId': output_folder_id,
+            'path': '.'
+        }]
+    else:
+        task.taskflow.logger.info('No output.folder.id')
+
+    upload_job_output_to_folder(cluster, job, log_write_url=None, job_dir=None,
+                                girder_token=task.taskflow.girder_token)
+
+    task.taskflow.logger.info('Upload complete.')
```

Comment:
Well we can trash the motorbike.py. That was my first test bed when simput was not yet involved.